pkg/kubeapiserver/authenticator/config.go

```diff
@@ -164,7 +164,7 @@ func (config AuthenticatorConfig) New() (authenticator.Request, *spec.SecurityDe
 		if err != nil {
 			return nil, nil, err
 		}
-		tokenAuthenticators = append(tokenAuthenticators, oidcAuth)
+		tokenAuthenticators = append(tokenAuthenticators, audunaware.WrapToken(config.ServiceAccountAPIAudiences, oidcAuth))
```

Comment:
it seems odd that the oidc authenticator won't authenticate for the audience contained in the id_token... I'd probably plumb implicit audiences into the oidc authenticator and let it accept the union of the implicit ones and the client_id

Reply:
Will drop this change and fix oidc authenticator in a better way.